app/models/work-item.ts

```diff
@@ -491,6 +498,32 @@ export async function deleteWorkItemsById(
   return numDeleted;
 }
 
+/**
+ * Compute the max sort index (used for batching) for the given job/service. This depends
+ * on the previous service executing one at a time, such as query-cmr, otherwise table locking
+ * or some other solution must be employed to ensure that simultaneous calls to this function
+ * don't return the same sort index.
+ *
+ * @param tx - the transaction to use for querying
+ * @param jobID - the ID of the job that created the work item
+ * @param serviceID - the serviceID of the step within the workflow
+ * @returns a promise containing the max stepIndex value or -1 if there are no matching rows
+ */
+export async function maxSortIndexForJobService(
+  tx: Transaction,
+  jobID: string,
+  serviceID: string,
+): Promise<number> {
+  const result = await tx(WorkItem.table)
+    .where({
+      jobID,
+      serviceID,
+    })
+    .max('sortIndex', { as: 'max' })
+    .first();
+  return result?.max || -1;
```

Comment:
if result.max == 0 this will return -1 instead of 0, right? Is that a problem?

Reply:
Good catch. Fixed.